src/solver/td3.ml

```diff
@@ -335,7 +375,8 @@ module Base =
           if not (Timing.wrap "S.Dom.equal" (fun () -> S.Dom.equal old wpd) ()) then ( (* value changed *)
             if tracing then trace "sol" "Changed";
             (* if tracing && not (S.Dom.is_bot old) && HM.mem wpoint x then trace "solchange" "%a (wpx: %b): %a -> %a" S.Var.pretty_trace x (HM.mem wpoint x) S.Dom.pretty old S.Dom.pretty wpd; *)
-            if tracing && not (S.Dom.is_bot old) && HM.mem wpoint x then trace "solchange" "%a (wpx: %b): %a" S.Var.pretty_trace x (HM.mem wpoint x) S.Dom.pretty_diff (wpd, old);
+            (* is there a reason for the redundant widen check? *)
+            if tracing && not (S.Dom.is_bot old) && should_widen x then trace "solchange" "%a (wpx: %b): %a" S.Var.pretty_trace x (should_widen x) S.Dom.pretty_diff (wpd, old);
```

Comment:
What is this comment about?

Reply:
My comment is strangely worded, but I think it is odd that `should_widen` or previously `HM.mem wpoint x` is re-evaluated here.
Specifically, `let wp = should_widen x` before evaluating `eq` is used to determine, whether widening should be applied. The result of `should_widen` may have changed through `eq`, yet I would assume the value that is logged should reflect the value that is used for the actual widening decision.

But I should definitely be logging with `format_wpoint`, which will not only show whether x is a `wpoint`, but also how much gas it has.